lms/djangoapps/experiments/utils.py

```diff
@@ -264,46 +264,118 @@ def get_dashboard_course_info(user, dashboard_enrollments):
 
 def get_experiment_user_metadata_context(course, user):
     """
-    Return a context dictionary with the keys used by the user_metadata.html.
+    Return a context dictionary with the keys used for Optimizely experiments, exposed via user_metadata.html
```

Comment:
Can we be more descriptive with what the expected output should be? For example, listing the keys and a high level description of the expected information for each key?

Reply:
A lot of these fields are super common in lms, but sometimes seeing sample values is the quickest way to understand the context, especially when some terms get repurposed incorrectly in the actual code (*cough* course key *cough*). 

Since the comments mention that it's exposed via user_metadata.html, I've added a note with the command to view it, so anyone unfamiliar with user-metadata can see it first-hand. That way they can see the exact values. It may be beneficial down the road to have a descriptive glossary the metadata fields (and probably these lms terms in general), but I think that level of documentation-backfill is beyond the scope of the ticket at hand (to expose this existing data to an endpoint for our internal use).

Reply:
I was thinking something like this so it's a little more in line with more modern commenting styles: 
```
Retrieves user + course data used to enrich Optimizely experiments.

Args:
  course - course_key
  user - username of the currently logged in user

Returns:
  {
   has_non_audit_enrollments: {},
   has_staff_access: boolean, 
   forum_roles: [],
   partition_groups: [],
   user_metadata: {}
  }
```

With the return values' keys like this, the next person to look at this doesn't need to either trigger the function to figure out the top level keys or read through the entire function to do so. And if there are things that are confusing, it's might be worth documenting in the description so the next person doesn't need to go through the same troubles you did.

Reply:
Sorry I think my summary was confusing: I was actually suggesting that seeing actual example values gives faster/clearer context (to me at least) than saying an element will contain the "course key"; that in my travels with Matthew it sounds like some of these terms themselves have morphed and created confusion, as a developer getting to this part what I really want to know is "is this value the one that looks like the concatenated string? or the one with three parts?" 

To be clear, I agree your format is the normal way to go about this (for normal data), but in this particular case I think the reader will actually get more definitive info from specific values than a short description. If there's developer confusion around these user-metadata fields it may be worth backfilling some of the function comments to describe/differentiate the items in this dictionary, but I think that's beyond the scope of this ticket.